client/src/bidirectional_channel_handler.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::{Duration, Instant},
+};
+
+use crossbeam_channel::{Receiver, Sender};
+use quinn::RecvStream;
+use solana_streamer::bidirectional_channel::QuicReplyMessage;
+
+// This structure will handle the bidirectional messages that we get from the quic server
+// It will save 1024 QuicReplyMessages sent by the server in the crossbeam receiver
+// This class will also handle recv channel created by the QuicClient when connecting to the server in bidirectional mode
+#[derive(Clone)]
+pub struct BidirectionalChannelHandler {
+    sender: Arc<Sender<QuicReplyMessage>>,
+    pub reciever: Receiver<QuicReplyMessage>,
+    recv_channel_is_set: Arc<AtomicBool>,
+}
+
+impl BidirectionalChannelHandler {
+    pub fn new() -> Self {
+        let (sender, reciever) = crossbeam_channel::bounded(1024);
```

Comment:
1) why is this channel bounded?
2) what happens when the limit is reached?

Reply:
So that we add more messages only if client is using these messages. We cannot add more messages and here we silently continue. I can make it unbounded for the moment.

Reply:
Made it unbonded.